### ISSUE TITLE
Address cdms2 slow read from multiple netcdf files accessed through spanning xml

### ISSUE THREAD

**Issue by durack1** [UV-CDAT/uvcdat#479]
It appears cdms2 is taking a very long time (tens of minutes) to read a subset of data from multiple netcdf files containing a continuous time axis. In particular the issue is appearing with very high resolution (daily, 2700 x 3600) surface ocean fields.

The slow read also happens when addressing the netcdf file(s) directly

The issue is apparent on the GFDL computing systems and has also been replicated on a workstation physical disk so it's likely a cdms2 problem rather than anything hardware related.

**Comment by doutriaux1** [comment]
@painter1 that is the bug I'm thinking is hitting us on rhea. Will take a look tomorrow

**Comment by doutriaux1** [comment]
@durack1 are the files still around? @painter1 thread seems to indicate -j helps. Can't reproduce locally on a tiny example.

**Comment by painter1** [comment]
When I looked at the -j option it didn't seem that it does much of anything relative to the default behavior, except that it disables the default behavior of linearizing time for very short time units.  Maybe I missed something.

**Comment by durack1** [comment]
@doutriaux1 yep - ocean:/export/doutriaux1/Paul we last discussed this back in June last year..

**Comment by durack1** [comment]
@doutriaux1 can you assign the enhancement label please?

**Comment by doutriaux1** [comment]
@dnadeau4 assigning this to you but let's work on it together

**Comment by dnadeau4** [comment]
I think it affect using THREDDS as well.  #1475  I would like to only read 1 time dimension at a time instead of loading everything in memory at once.    Will start to investigate a solution.

**Comment by durack1** [comment]
@dnadeau4 happy to run any tweaks you have over the test case, we have it pretty well reproduced..

**Comment by dnadeau4** [comment]
I have made progress on this parallel issue.   Everything is working well.   I an parallel world, every node must open the file, create dimensions and create variables.  Only slicing is different for each rank.

Please run testing/cmds2/test_mpi_write_2.py  on branch issue_479.

**Comment by durack1** [comment]
@dnadeau4 are the tweaks on the branch above addressing the slow read times which was the focus of this original issue?

**Comment by dnadeau4** [comment]
This is a write in parallel program.  Nothing to do with the original issue. 

The original issue will need a re-architecture of CDMS.  CDMS read the entire array in memory.  I would like to change the code just to return a empty handler and read slices of the array per user request.  Of course if the user ask for the entire array it will still be slow, but it could be possible to read in parallel using multiple nodes. Something to think about.

**Comment by durack1** [comment]
@dnadeau4 if there is no "fix" for this issue, aside from a rearchitecture of `cdms2`, then close this issue and add the large grid test to the suite for testing when `cdms2` is updated..
